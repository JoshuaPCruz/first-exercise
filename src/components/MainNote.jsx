import React from 'react';
import '../stylesheets/components/MainNote.css';
import obama from '../img/newYork.png';

const MainNote = () => {
  return (
    <>
      <div className="mainNote">
        <div className="imgContainer">
          <img src={obama} alt="Obama" />
        </div>
        <h3>Rammury and Obama Bristle From Start</h3>
        <p>
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Tenetur
          facere eum vero molestias culpa adipisci aut harum distinctio
          voluptates laudantium veritatis minima expedita autem natus eos
          eius dolore, quae aspernatur.
            </p>
      </div>
      <div className="noteDetails">
        <h2>Lorem ipsum dolor sit amet consectetur adipisicing elit</h2>
        <div className="noteDetails__block">
          <div className="separator"></div>
          <h3>Some title</h3>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Animi
            repudiandae, ex excepturi illo deserunt officiis alias. Facilis
            deleniti ipsam quaerat architecto, sed perferendis molestias
              </p>
        </div>
        <div className="noteDetails__block">
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quae
            tempore aperiam itaque, ab explicabo, nulla laboriosam cum ipsum
            eveniet enim minima ut consequuntur perferendis aut. Suscipit
            dignissimos quas est odio?
              </p>
        </div>
        <div className="noteDetails__block">
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quae
            tempore aperiam itaque, ab explicabo, nulla laboriosam cum ipsum
            eveniet enim minima ut consequuntur perferendis aut. Suscipit
            dignissimos quas est odio?
              </p>
        </div>
        <div className="noteDetails__block">
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quae
            tempore aperiam itaque, ab explicabo, nulla laboriosam cum ipsum
            eveniet enim minima ut consequuntur perferendis aut. Suscipit
            dignissimos quas est odio?
              </p>
        </div>
      </div>
    </>
  );
};

export default MainNote;