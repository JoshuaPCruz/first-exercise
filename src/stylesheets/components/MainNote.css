.mainNote {
  border-bottom: 2px solid var(--color-primary-black);
  padding-bottom: 0.5em;
}

.noteDetails {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: 27px 1fr;
  grid-template-areas: "title title title title";
  grid-gap: 1em;
  padding: 1em 0;
  border-bottom: 2px solid var(--color-primary-black);
}

.imgContainer img {
  width: 100%;
}

.mainNote h3 {
  margin: 0.2em 0;
}

.noteDetails h2 {
  grid-area: title;
  font-style: italic;
  font-size: 1.9em;
}

.noteDetails__block .separator {
  margin-top: 0;
}

.noteDetails__block h3 {
  margin-bottom: 0.8em;
  text-align: center;
}