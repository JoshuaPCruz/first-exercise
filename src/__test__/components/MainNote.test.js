import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import MainNote from '../../components/MainNote';

describe('<MainNote />', () => {
  const mainNote = shallow(<MainNote />);
  test('Render del componente MainNote', () => {
    expect(mainNote.length).toEqual(1);
  });

  test('Render del tÃ­tulo', () => {
    expect(mainNote.find('h3').first().text()).toEqual('Rammury and Obama Bristle From Start')
  });

});

describe('MainNote Snapshot', () => {
  test('Comprobar la UI del componente MainNote', () => {
    const mainNote = create(<MainNote />);
    expect(mainNote.toJSON()).toMatchSnapshot();
  });
});