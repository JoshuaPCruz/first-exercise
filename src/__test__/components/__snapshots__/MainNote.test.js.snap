// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNote Snapshot Comprobar la UI del componente MainNote 1`] = `
Array [
  <div
    className="mainNote"
  >
    <div
      className="imgContainer"
    >
      <img
        alt="Obama"
        src={Object {}}
      />
    </div>
    <h3>
      Rammury and Obama Bristle From Start
    </h3>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Tenetur facere eum vero molestias culpa adipisci aut harum distinctio voluptates laudantium veritatis minima expedita autem natus eos eius dolore, quae aspernatur.
    </p>
  </div>,
  <div
    className="noteDetails"
  >
    <h2>
      Lorem ipsum dolor sit amet consectetur adipisicing elit
    </h2>
    <div
      className="noteDetails__block"
    >
      <div
        className="separator"
      />
      <h3>
        Some title
      </h3>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Animi repudiandae, ex excepturi illo deserunt officiis alias. Facilis deleniti ipsam quaerat architecto, sed perferendis molestias
      </p>
    </div>
    <div
      className="noteDetails__block"
    >
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Quae tempore aperiam itaque, ab explicabo, nulla laboriosam cum ipsum eveniet enim minima ut consequuntur perferendis aut. Suscipit dignissimos quas est odio?
      </p>
    </div>
    <div
      className="noteDetails__block"
    >
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Quae tempore aperiam itaque, ab explicabo, nulla laboriosam cum ipsum eveniet enim minima ut consequuntur perferendis aut. Suscipit dignissimos quas est odio?
      </p>
    </div>
    <div
      className="noteDetails__block"
    >
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Quae tempore aperiam itaque, ab explicabo, nulla laboriosam cum ipsum eveniet enim minima ut consequuntur perferendis aut. Suscipit dignissimos quas est odio?
      </p>
    </div>
  </div>,
]
`;
